packages/@sanity/desk-tool/src/panes/document/statusBar/sparkline/DocumentSparkline.tsx

```diff
@@ -1,74 +1,85 @@
+import {useSyncState} from '@sanity/react-hooks'
 import {useDocumentPane} from '../../useDocumentPane'
 import {DocumentBadges} from './DocumentBadges'
-import {ReviewChangesButton} from './ReviewChangesButton'
-import {IconBadge} from './IconBadge'
-import {
-  BadgesBox,
-  MetadataBox,
-  ReviewChangesBadgeBox,
-  ReviewChangesButtonBox,
-} from './DocumentSparkline.styled'
 import {PublishStatus} from './PublishStatus'
+import {ReviewChangesButton} from './ReviewChangesButton'
 
 export const DocumentSparkline = memo(function DocumentSparkline() {
-  const {editState, historyController, value} = useDocumentPane()
+  const {
+    editState,
+    historyController,
+    value,
+    documentId,
+    documentType,
+    handleHistoryClose,
+    handleHistoryOpen,
+    changesOpen,
+  } = useDocumentPane()
+  const syncState = useSyncState(documentId, documentType)
+
   const lastUpdated = value?._updatedAt
   const lastPublished = editState?.published?._updatedAt
   const showingRevision = historyController.onOlderRevision()
   const liveEdit = Boolean(editState?.liveEdit)
   const published = Boolean(editState?.published)
   const changed = Boolean(editState?.draft)
-  const loaded = published || changed
 
-  // Keep track of the size of the review changes button
-  const [
-    reviewChangesButtonElement,
-    setReviewChangesButtonElement,
-  ] = useState<HTMLButtonElement | null>(null)
-  const reviewChangesButtonElementRect = useElementRect(reviewChangesButtonElement)
-  const reviewChangesButtonWidth = reviewChangesButtonElementRect?.width
-  const reviewChangesButtonWidthRef = useRef(reviewChangesButtonWidth || 0)
+  const [rootFlexElement, setRootFlexElement] = useState<HTMLDivElement | null>(null)
+  const rootFlexRect = useElementRect(rootFlexElement)
+  const collapsed = Boolean(rootFlexRect && rootFlexRect?.width < 300)
 
-  // Use the size of the review changes button as a element query breakpoint
-  const metadataBoxBreakpoints = useMemo(
-    () => [reviewChangesButtonWidth || reviewChangesButtonWidthRef.current, 225],
-    [reviewChangesButtonWidth]
-  )
+  const [status, setStatus] = useState<'saved' | 'syncing' | null>(null)
 
-  // Only transition between subsequent state, not the initial
-  const [transition, setTransition] = useState(false)
+  // eslint-disable-next-line consistent-return
   useEffect(() => {
-    if (!transition && loaded) {
-      // NOTE: the reason for double RAF here is a common "bug" in browsers.
-      // See: https://stackoverflow.com/questions/44145740/how-does-double-requestanimationframe-work
-      // There is no need to cancel this animation,
-      // since calling it again will cause the same result (transition=true).
-      return raf2(() => setTransition(true))
+    if (status === 'syncing') {
+      // status changed to 'syncing', schedule an update to set it to 'saved'
+      const timerId = setTimeout(() => setStatus('saved'), 1000)
```

Comment:
Idea: the `1000` constant could be named to `SYNC_SAVE_TIMEOUT` and placed in the global scope.